extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/DeltaSession.java

```diff
@@ -586,6 +610,20 @@ public int getSizeInBytes() {
     return serializedAttributes;
   }
 
+  protected ConcurrentMap getAttributes() {
+    try {
+      Field field = getAttributesFieldObject();
+      field.setAccessible(true);
+      Map oldMap = (Map) field.get(this);
+      ConcurrentMap newMap = new ConcurrentHashMap();
```

Comment:
Why do we need to copy the map here? I think the attributes field is always set to a ConcurrentHashMap, even though in tomcat 6 the field was of type map.